bin/build-darwin.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# This script buildd BlazingMQ and all of its dependencies.
+#
+# Required prerequisites:
+# - git
+# - Perl
+# - Python3
+# - Clang
+# - CMake
+# - pkg-config
+# - Homebrew
+
+set -e
+set -u
+set -x
+[ -z $BASH ] || shopt -s expand_aliases
+
+
+# :: Set some initial constants :::::::::::::::::::::::::::::::::::::::::::::::
+DIR_ROOT="${DIR_ROOT:-`pwd`}"
+DIR_THIRDPARTY="${DIR_ROOT}/thirdparty"
+
+
+# :: Download source-code of our dependencies :::::::::::::::::::::::::::::::::
+if [ ! -d "${DIR_ROOT}/blazingmq" ]; then
+    git clone https://github.com/bloomberg/blazingmq.git
+fi
+if [ ! -d "${DIR_THIRDPARTY}/bde-tools" ]; then
+    git clone https://github.com/bloomberg/bde-tools "${DIR_THIRDPARTY}/bde-tools"
+fi
+if [ ! -d "${DIR_THIRDPARTY}/bde" ]; then
+    git clone https://github.com/bloomberg/bde.git "${DIR_THIRDPARTY}/bde"
+fi
+if [ ! -d "${DIR_THIRDPARTY}/ntf-core" ]; then
+    git clone https://github.com/bloomberg/ntf-core.git "${DIR_THIRDPARTY}/ntf-core"
+fi
+
+DIR_BUILD="${DIR_ROOT}/blazingmq/cmake.bld/`uname`"
+mkdir -p ${DIR_BUILD}
+
+
+# :: Install required packages ::::::::::::::::::::::::::::::::::::::::::::::::
+
+# Build and install BDE
+# Refer to https://bloomberg.github.io/bde/library_information/build.html
+PATH="${DIR_THIRDPARTY}/bde-tools/bin:$PATH"
+
+pushd ${DIR_THIRDPARTY}/bde
+eval `bbs_build_env -u opt_64_cpp17 -b "${DIR_BUILD}/bde"`
+bbs_build configure --prefix="${DIR_ROOT}"
+bbs_build build --prefix=""${DIR_ROOT}
+bbs_build --install_dir="/" --prefix="${DIR_ROOT}" install
+eval `bbs_build_env unset`
+popd
+
+# Build and install NTF
+pushd ${DIR_THIRDPARTY}/ntf-core
+./configure --prefix "${DIR_ROOT}" --output "${DIR_BUILD}/ntf"
+make
+make install
+popd
+
+# Build other dependencies
+brew install flex bison google-benchmark zlib
+
+
```

Comment:
Something like this will ensure work with `.local` hostname, but might be good to explicitly ask the user if they want to update `hosts`
```suggestion
SELF_HOST_ADDRESS="127.0.0.1       `hostname`"
if ! (grep -q "$SELF_HOST_ADDRESS" /etc/hosts > /dev/null);
then
    sudo bash -c "echo $SELF_HOST_ADDRESS >> /etc/hosts"
fi
```

Reply:
This is updating `/etc/hosts` and using `sudo`

Might be better to explicitly ask the user

Reply:
Let's just propose then, so user decides:

```suggestion

SELF_HOST_ADDRESS="127.0.0.1       `hostname`"
if ! (grep -q "$SELF_HOST_ADDRESS" /etc/hosts > /dev/null);
then
    echo "Warning: self hostname `hostname` not found in /etc/hosts"
    echo "It might be necessary to add it manually to launch bmqbrkr:"
    echo "sudo bash -c \"echo \\\"$SELF_HOST_ADDRESS\\\" >> /etc/hosts\""
fi

```

The output:
```
% ./tmp.sh
Warning: self hostname MYHOST not found in /etc/hosts
It might be necessary to add it manually to launch bmqbrkr:
sudo bash -c "echo \"127.0.0.1       MYHOST\" >> /etc/hosts"
```